sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java

```diff
@@ -132,10 +137,38 @@ public void verifyOnline(byte[] artifactDigest, byte[] certChain, byte[] signatu
     }
   }
 
-  public void verify(KeylessVerificationRequest request) throws KeylessVerificationException {
+  /** Convenience wrapper around {@link #verify(byte[], KeylessVerificationRequest)}. */
+  public void verify(Path artifact, KeylessVerificationRequest request)
+      throws KeylessVerificationException {
+    try {
+      byte[] artifactDigest =
+          Files.asByteSource(artifact.toFile()).hash(Hashing.sha256()).asBytes();
+      verify(artifactDigest, request);
+    } catch (IOException e) {
+      throw new KeylessVerificationException("Could not hash provided artifact path: " + artifact);
+    }
+  }
+
+  /**
+   * Verify that the inputs can attest to the validity of a signature using sigstore's keyless
+   * infrastructure. If no exception is thrown, it should be assumed verification has passed.
+   *
+   * @param artifactDigest the sha256 digest of the artifact that is being verified
+   * @param request the keyless verification data and options
+   * @throws KeylessVerificationException if the signing information could not be verified
+   */
+  public void verify(byte[] artifactDigest, KeylessVerificationRequest request)
+      throws KeylessVerificationException {
     var signingCert = SigningCertificate.from(request.getKeylessSignature().getCertPath());
     var leafCert = signingCert.getLeafCertificate();
 
+    // this ensures the provided artifact digest matches what may have come from a bundle (in
+    // keyless signature)
+    if (!Arrays.equals(artifactDigest, request.getKeylessSignature().getDigest())) {
+      throw new KeylessVerificationException(
+          "Provided artifact digest does not match digest used for verification");
```

Comment:
We should probably list both hashes for the user to analyze

Reply:
done